routing/blinding.go

```diff
@@ -0,0 +1,69 @@
+package routing
+
+import (
+	"errors"
+	"fmt"
+
+	sphinx "github.com/lightningnetwork/lightning-onion"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+var (
+	// ErrNoBlindedPath is returned when the blinded path in a blinded
+	// payment is missing.
+	ErrNoBlindedPath = errors.New("blinded path required")
+
+	// ErrInsufficientBlindedHops is returned when a blinded path does
+	// not have enough blinded hops.
+	ErrInsufficientBlindedHops = errors.New("blinded path requires " +
+		"at least one hop")
+)
+
+// BlindedPayment provides the path and payment parameters required to send a
+// payment along a blinded path.
+type BlindedPayment struct {
```

Comment:
We already have a [CLTVLimit](https://github.com/lightningnetwork/lnd/blob/f1bca2a59c6117cd2a23895d6b9c9a339904468b/routing/pathfind.go#L336) in our routing restrictions, so I was thinking that we just re-use that? 

There is a bit of a design choice in here about how we want to deal with the split between blinded payments and regular ones where there is field overlap: 

1. Only add additional blinded payment info, then re-use common fields (as is done here)
2. Have a completely separate set of fields, ensure they are mutually exclusive then merge

I've found (1) to be a _little_ more simple when it gets down the the API design for these (two fields in different places are confusing), so that's what swayed me in the end.